### ISSUE TITLE
Allow list arguments in default files to accumulate

### ISSUE THREAD

**Issue by jgm** [jgm/pandoc#5881]
For example, if defaults1.yaml is
```
include-in-header: A.txt
```
and defaults2.yaml is
```
include-in-header:
- B.txt
- C.txt
```
and the command line is
```
pandoc --include-in-header D.txt --defaults defaults1.yaml --defaults defaults2.yaml
```
then we should get four files included, in the order: D.txt, A.txt, B.txt, C.txt.

This affects many of the arguments in defaults files.  Note that some of these are currently accumulated in reverse in option processing (when multiple options are allowed), and we may want to regularize this a bit.

We also need some tests for this sort of thing.

**Comment by jgm** [comment]
This may also help with #3139

**Comment by dhimmel** [comment]
What is the difference between `--defaults` and `--metadata-file`?

**Comment by brainchild0** [comment]
In the past when I have dealt with problems analogous to this one I have found that trying to apply a single a rule to all fields of a common basic type (e.g. lists) is generally prone to lead to confusing boundary conditions.  Some list fields might be suitable handling through a list append, as you have shown, some by pre-pending, some by full replacement, and some by a case-by-case basis depending on values of other fields.  It is a complex subject that is tempting to try to treat generically, though doing so is not always viable.

**Comment by jgm** [comment]
Closed by ec043e0d97fdc864044f7c9e7efaa1b4c733bafb
